### ISSUE TITLE
No Response since Pentair iOS app update

### ISSUE THREAD

**Issue by kahnartist81** [dustindclark/homebridge-pentair-intellicenter#48]
Like the title says, since Pentair updated their iOS app, none of my pool features work via HOOBS/Homekit. At first it was just my pool circuit, then when I tried to check all of my credentials, everything else No Response. I've tried downgrading the plugin, uninstalling, creating a new bridge, all to no avail.Seems to be broken.

iOS 15.7, 16.0.3 iOS 16.1
IntelliCenter2 app version 2.5.36
Homebridge-pentair-intellicenter v2.0.3
HOOBS v 4.2.8

**Comment by dustindclark** [comment]
Did you try rebooting your Intellicenter panel? Which firmware is it running?

**Comment by kahnartist81** [comment]
﻿I have no tried the panel yet. Figured it was ok since it works perfectly.fine through it's own app. It's on v 1.064 firmware. Rebooting panel now.--eric ***@***.*** Oct 27, 2022, at 4:44 PM, Dustin Clark ***@***.***> wrote:﻿
Did you try rebooting your Intellicenter panel? Which firmware is it running?

**Comment by kahnartist81** [comment]
No luck. Restarted panel and plugin bridge and home app. Still getting no response after "updating" prompt.--eric ***@***.*** Oct 27, 2022, at 4:44 PM, Dustin Clark ***@***.***> wrote:﻿
Did you try rebooting your Intellicenter panel? Which firmware is it running?

**Comment by kahnartist81** [comment]
Just updated to 2.0.4 for the plugin, and no change. Still not responding.--eric ***@***.*** Oct 27, 2022, at 4:44 PM, Dustin Clark ***@***.***> wrote:﻿
Did you try rebooting your Intellicenter panel? Which firmware is it running?

**Comment by dustindclark** [comment]
I'm on the same version of everything and can't recreate. The only thing that I can think of is that somehow you have 2 instances of the plugin and/or homebridge running, which will break because the IntelliCenter port is in use. I would try 2 things:

1. Enable homebridge debug logging to see if you get more descriptive errors.
2. Reboot the OS that your homebridge is running on. Not just the homebridge process...the full OS.

**Comment by kahnartist81** [comment]
So weird. I'm running Hoobs on a RPi and have restarted a few times already hoping that would work. Here's a screenshot of my logs after I updated to 2.0.4. The gas heater error repeats until current time.--eric ***@***.*** Oct 28, 2022, at 3:20 PM, Dustin Clark ***@***.***> wrote:﻿
I'm on the same version of everything and can't recreate. The only thing that I can think of is that somehow you have 2 instances of the plugin and/or homebridge running, which will break because the IntelliCenter port is in use. I would try 2 things:

Enable homebridge debug logging to see if you get more descriptive errors.
Reboot the OS that your homebridge is running on. Not just the homebridge process...the full OS.

**Comment by dustindclark** [comment]
Can't see the logs. You'll have to post on GitHub.

**Comment by kahnartist81** [comment]
Sorry about. I posted an image. I seem.to have gotten it mostly working. All of the switches work. The heaters show the set temperature. But both the pool and spa are reading 32' whereas the Pentair app shows them at 66 and 64. Here are the logs below as text from after the reinstall.10/28/2022, 3:43:00 PMPentair Intellicenter BridgeBridge change10/28/2022, 3:43:11 PMPentair Intellicenter BridgeLoaded plugin 'homebridge-pentair-intellicenter'10/28/2022, 3:43:11 PMPentair Intellicenter BridgeRegistering platform 'homebridge-pentair-intellicenter.PentairIntelliCenter'10/28/2022, 3:43:11 PMPentair Intellicenter BridgeLoading 1 platforms...10/28/2022, 3:43:12 PMPentair Intellicenter BridgeBridge is running on port 51836.10/28/2022, 3:43:13 PMPentair Intellicenter BridgeERROR(node:2015) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit10/28/2022, 3:43:13 PMPentair Intellicenter BridgeERROR(Use `node --trace-warnings ...` to show where the warning was created)10/28/2022, 3:43:13 PMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 18.333333333333332 exceeded minimum of 21.1111111111111110/28/2022, 3:43:13 PMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.77777777777778 exceeded minimum of 21.1111111111111110/28/2022, 3:43:33 PMPentair Intellicenter BridgeLoaded plugin 'homebridge-pentair-intellicenter'10/28/2022, 3:43:33 PMPentair Intellicenter BridgeLoading 1 platforms...10/28/2022, 3:43:33 PMPentair Intellicenter BridgeRegistering platform 'homebridge-pentair-intellicenter.PentairIntelliCenter'10/28/2022, 3:43:33 PMPentair Intellicenter BridgeBridge is running on port 51836.10/28/2022, 3:43:34 PMPentair Intellicenter BridgeERROR(node:2156) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit10/28/2022, 3:43:34 PMPentair Intellicenter BridgeERROR(Use `node --trace-warnings ...` to show where the warning was created)10/28/2022, 3:45:23 PMPentair Intellicenter BridgePentairIntelliCenterSetting Pool to true10/28/2022, 3:46:05 PMPentair Intellicenter BridgePentairIntelliCenterSet heat power to 1 for heater Gas Heater10/28/2022, 3:46:07 PMPentair Intellicenter BridgePentairIntelliCenterSet heat power to 0 for heater Gas Heater10/28/2022, 3:46:16 PMPentair Intellicenter BridgePentairIntelliCenterSetting Pool to false--eric ***@***.*** Oct 28, 2022, at 3:31 PM, Dustin Clark ***@***.***> wrote:﻿
Can't see the logs. You'll have to post on GitHub.

**Comment by dustindclark** [comment]
Is the outside temp 32 where you are? The water has to be flowing for IntelliCenter to get a reading. Try turning on the pump.

**Comment by kahnartist81** [comment]
It is not. The temp is 70' where I am. I turned on the pump for about 1-2 minutes hoping it would update but it did not. Thought maybe the logs would show something why...--eric ***@***.*** Oct 28, 2022, at 4:00 PM, Dustin Clark ***@***.***> wrote:﻿
Is the outside temp 32 where you are? The water has to be flowing for IntelliCenter to get a reading. Try turning on the pump.

**Comment by kahnartist81** [comment]
Just wanted to let you know that it stopped working again, and after a restart it's still no response. Here's the logs after the restart.10/29/2022, 10:25:08 AMPentair Intellicenter BridgeLoaded plugin 'homebridge-pentair-intellicenter'10/29/2022, 10:25:08 AMPentair Intellicenter BridgeLoading 1 platforms...10/29/2022, 10:25:08 AMPentair Intellicenter BridgeRegistering platform 'homebridge-pentair-intellicenter.PentairIntelliCenter'10/29/2022, 10:25:09 AMPentair Intellicenter BridgeBridge is running on port 51836.10/29/2022, 10:25:10 AMPentair Intellicenter BridgeERROR(node:1736) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit10/29/2022, 10:25:10 AMPentair Intellicenter BridgeERROR(Use `node --trace-warnings ...` to show where the warning was created)10/29/2022, 10:25:14 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:25:14 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.77777777777778 exceeded minimum of 21.1111111111111110/29/2022, 10:25:29 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:25:36 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:25:44 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:25:53 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:25:57 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:26:05 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:26:24 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:26:29 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:26:44 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:26:49 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:27:16 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:27:32 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:27:45 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:27:52 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:28:04 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:28:20 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:28:24 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:28:38 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:28:41 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:28:44 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:28:52 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:28:57 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:29:06 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:29:14 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:29:18 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.1111111111111110/29/2022, 10:29:27 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 16.666666666666668 exceeded minimum of 21.1111111111111110/29/2022, 10:29:30 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 17.22222222222222 exceeded minimum of 21.11111111111111--eric ***@***.*** Oct 28, 2022, at 4:00 PM, Dustin Clark ***@***.***> wrote:﻿
Is the outside temp 32 where you are? The water has to be flowing for IntelliCenter to get a reading. Try turning on the pump.

**Comment by dustindclark** [comment]
This seems suspect: 
```
Pentair Intellicenter BridgeERROR(node:1736) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit
```
Only 8 listeners are created in code, and this only happens in the constructor...so it seems like the platform constructor is called twice. I would check that you don't have duplicate config. If you don't, it seems like there may be an issue with your Homebridge installation.

**Comment by kahnartist81** [comment]
Thanks for writing back. I went ahead and fully deleted the plugin and corresponding bridge. Then fully reinstalled the plugin with a new bridge. It's still not working correctly in the home app. BUT. The plugin does seem to work perfectly fine from the accessories section of HOOBS, with the exception of it still staying on 32' for both pool and spa. But I'm fully able to control it all within Hoobs, but in the home app itself, it's still "not responding."--eric ***@***.*** Nov 10, 2022, at 11:14 AM, Dustin Clark ***@***.***> wrote:﻿
This seems suspect:
Pentair Intellicenter BridgeERROR(node:1736) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit

Only 8 listeners are created in code, and this only happens in the constructor...so it seems like the platform constructor is called twice. I would check that you don't have duplicate config. If you don't, it seems like there may be an issue with your Homebridge installation.

**Comment by dustindclark** [comment]
It's possible that you're seeing old/cached accessories in the home app in that case. It's worth trying to delete the cached accessories, then restart to see if that clears things up.

**Comment by kahnartist81** [comment]
Tried what you said. Even tried switching the degree from F to C, then suddenly they showed correctly. Then I switch back to F. Restarted the app and back to total no response. Logs are also still showing 11 listeners.11/11/2022, 8:10:25 AMPentair Intellicenter BridgeLoaded plugin 'homebridge-pentair-intellicenter'11/11/2022, 8:10:25 AMPentair Intellicenter BridgeLoading 1 platforms...11/11/2022, 8:10:25 AMPentair Intellicenter BridgeRegistering platform 'homebridge-pentair-intellicenter.PentairIntelliCenter'11/11/2022, 8:10:26 AMPentair Intellicenter BridgeBridge is running on port 51836.11/11/2022, 8:10:27 AMPentair Intellicenter BridgeERROR(node:4185) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit11/11/2022, 8:10:27 AMPentair Intellicenter BridgeERROR(Use `node --trace-warnings ...` to show where the warning was created)11/11/2022, 8:13:06 AMPentair Intellicenter BridgePentairIntelliCenterSetting Pool to true11/11/2022, 8:13:50 AMPentair Intellicenter BridgePentairIntelliCenterSetting Pool to false11/11/2022, 8:13:54 AMPentair Intellicenter BridgePentairIntelliCenterSetting Water Feature to true11/11/2022, 8:14:08 AMPentair Intellicenter BridgePentairIntelliCenterSetting Water Feature to false11/11/2022, 8:15:27 AMPentair Intellicenter BridgeConfiguration change11/11/2022, 8:15:38 AMPentair Intellicenter BridgeLoaded plugin 'homebridge-pentair-intellicenter'11/11/2022, 8:15:38 AMPentair Intellicenter BridgeLoading 1 platforms...11/11/2022, 8:15:38 AMPentair Intellicenter BridgeRegistering platform 'homebridge-pentair-intellicenter.PentairIntelliCenter'11/11/2022, 8:15:38 AMPentair Intellicenter BridgeBridge is running on port 51836.11/11/2022, 8:15:40 AMPentair Intellicenter BridgeERROR(node:4573) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit11/11/2022, 8:15:40 AMPentair Intellicenter BridgeERROR(Use `node --trace-warnings ...` to show where the warning was created)11/11/2022, 8:15:41 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 59 exceeded minimum of 7011/11/2022, 8:15:45 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 59 exceeded minimum of 7011/11/2022, 8:15:58 AMPentair Intellicenter BridgeConfiguration change11/11/2022, 8:16:09 AMPentair Intellicenter BridgeLoaded plugin 'homebridge-pentair-intellicenter'11/11/2022, 8:16:09 AMPentair Intellicenter BridgeLoading 1 platforms...11/11/2022, 8:16:09 AMPentair Intellicenter BridgeRegistering platform 'homebridge-pentair-intellicenter.PentairIntelliCenter'11/11/2022, 8:16:09 AMPentair Intellicenter BridgeBridge is running on port 51836.11/11/2022, 8:16:11 AMPentair Intellicenter BridgeERROR(node:4694) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Telnet]. Use emitter.setMaxListeners() to increase limit11/11/2022, 8:16:11 AMPentair Intellicenter BridgeERROR(Use `node --trace-warnings ...` to show where the warning was created)11/11/2022, 8:16:11 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 15 exceeded minimum of 7011/11/2022, 8:16:11 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 35 exceeded minimum of 7011/11/2022, 8:16:12 AMPentair Intellicenter BridgeERROR[Pentair Intellicenter Bridge ***@***.*** Gas Heater@@Current Temperature] characteristic was supplied illegal value: number 15 exceeded minimum of 21.11111111111111--eric ***@***.*** Nov 10, 2022, at 11:57 AM, Dustin Clark ***@***.***> wrote:﻿
It's possible that you're seeing old/cached accessories in the home app in that case. It's worth trying to delete the cached accessories, then restart to see if that clears things up.

**Comment by dustindclark** [comment]
The configured temperature units for the plugin needs to match the temperature units configured in IntelliCenter. Meaning, both should be F or both should be C. Double check that they match.

**Comment by kahnartist81** [comment]
Holy crap I think I just fixed it. I had the minimum temperature set to 70, but the current temp is 59. So it was crashing the system. I lowered it to 50. And boom. It works! --eric ***@***.*** Nov 10, 2022, at 11:57 AM, Dustin Clark ***@***.***> wrote:﻿
It's possible that you're seeing old/cached accessories in the home app in that case. It's worth trying to delete the cached accessories, then restart to see if that clears things up.

**Comment by dustindclark** [comment]
Yep. That makes sense, and I see the bug. Will push a fix.

**Comment by kahnartist81** [comment]
Thanks for the help. Glad I was able to fix and expose a bug. Back to loving the plugin again. Haha. Keep up the good work.--eric ***@***.*** Nov 11, 2022, at 8:27 AM, Dustin Clark ***@***.***> wrote:﻿
Yep. That makes sense, and I see the bug. Will push a fix.

**Comment by dustindclark** [comment]
Fixed and published in 2.0.5. Closing.
